### PR TITLE
Add framework for Terra contract testing

### PR DESCRIPTION
**Pull request by karlkempe**
Spoke with @evan-gray about adding tests similar to those in the ethereum directory.

The purpose is to conduct various contract tests by deploying and interacting with these contracts in a local terra node. I've only written a few tests to start using Jest.

This follows the same format @kcsongor did using a Makefile (so all you would have to do is run "make test").

---
This change is [Reviewable](https://reviewable.io/reviews/certusone/wormhole/1050)

### CODE REVIEW

File: terra/test/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+## Running Local Terra Node
+
+In order to run these tests, you need to have a local Terra node running. These tests are meant to be run using [LocalTerra](https://github.com/terra-money/LocalTerra). This requires [Docker Compose](https://docs.docker.com/compose/install/) to run. You can also run _terrad_ with the same set up Tilt uses (see configuration [here](../../devnet/terra-devnet.yaml)).
+
+## Build
+
+In the [terra root directory](../), run the following:
+```sh
+docker build -f Dockerfile.build -o artifacts .
```
Comment (kcsongor) [inline terra/test/README.md:11]: could you update these instructions (and the test below too) to use `make`?

Comment (karlkempe) [reply terra/test/README.md:11]: Good catch. I'll change this

File: terra/test/src/helpers/vaa.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    abi.encodeParameter("uint16", toChain).substring(2 + (64 - 4)) +
+    // fee
+    abi.encodeParameter("uint256", relayerFee).substring(2);
+
+  // additional payload
+  if (additionalPayload === undefined) {
+    additionalPayload = "";
+  }
+
+  // TODO: fix this
```
Comment (kcsongor) [inline terra/test/src/helpers/vaa.ts:53]: what is there to fix here?

Comment (karlkempe) [reply terra/test/src/helpers/vaa.ts:53]: I should have deleted this comment actually. There isn't anything to fix here anymore.

**Review by kcsongor — APPROVED** [review]
Looks good! 👍

**Review by kcsongor — APPROVED** [review]

**Review by evan-gray — APPROVED** [review]

**Pull request merged**